app/src/main/java/org/xwiki/android/sync/contactdb/ContactManager.kt

```diff
@@ -108,11 +109,14 @@ class ContactManager(
                         if (batchOperation.size() >= 100) {
                             batchOperation.execute()
                         }
-                        contactManager.updateAvatar(
-                            resolver,
-                            contactManager.lookupRawContact(resolver, xWikiUserFull.id),
-                            xWikiUserFull
-                        )
+
+                        if (!getValue(context, "data_saving", false)) {
```

Comment:
Will be better to create extension value for context, like

```
var Context.dataSaverModeEnabled: Boolean
    set(value) {
        // set in shared prefs
    }
    get() {
        getValue(this, "data_save_mode", false)
    }
```

And its usage everywhere

Reply:
Shall I put that in the appContext?